### DISCUSSION TITLE
Help with RDR2 install.

### DISCUSSION THREAD

**Discussion by DumbledoreCalrissian** [Heroic-Games-Launcher/HeroicGamesLauncher#2057] [Q&A] [answered]
RDR2 install spits out a 'Social Club failed to initialise graphics: code 18' error.
Multiple online forum posts point to adding a launch option (in steam) - "WINEDLLOVERRIDES=vulkan-1=n,b %command%" to fix the error.
How do I do this in Heroic?
I saw a Reddit post that pointed to an 'advanced options' where I can enter this. But this seems to be for an older version. 
Linux noob here. 
Thanks for your help!

**Comment by DumbledoreCalrissian** [comment, marked as answer]
I found it!
I added an environment variable like this -
Variable Name : WINEDLLOVERRIDES
Value: vulkan-1=n,b

**Reply by JSwierk** [reply]
I know the post is old, but I have a question. Do you have your game installed on Steam deck internal storage or SD card?

**Reply by DumbledoreCalrissian** [reply]
I'm not on the Steam Deck. I have on Lenovo Legion laptop. Why do you ask?

**Reply by JSwierk** [reply]
Oh sorry, made a mistake.

**Reply by CursedDani** [reply]
Sorry, complete Linux noob here. How and where do you add that variable?

**Reply by dp27thelight** [reply]
Hi I wasn't having any luck with Elden Ring using ModEngine2

In lutris I set DLL overrides to

`WINEDLLOVERRIDES="dinput8=n,b"`

I tried

Variable Name : WINEDLLOVERRIDES
Value:dinput8=n.b

Already works great with Lutris, but I'm trying to learn Heroic Game Launcher that has some nice features, but is also hit and miss in some areas.

**Comment by Giger22** [comment]
Thanks!

**Comment by makimanson** [comment]
Hi!

Although this post is a bit old, I'd like to share my experience.

I've decided to do it here rather than create a new thread; I'm not sure if that's the right thing to do.

In any case, here it is.

In my case, the setup might be a bit unusual: I use a default prefix to install/import all my games, thus avoiding redundancies and, above all, saving disk space.
Also, to further elaborate, and in case anyone is interested, I redirect folders like: Documents, AppData (selectively), Save Games, Pictures... to retrieve saved games, settings, user profiles, etc., and avoid duplicates. All of this is done from NTFS drives. Everything is clean and transparent with more than 30 games installed/imported to Heroic from different platforms (Ubisoft, EA app, GOG, Battle.net), all working perfectly.
I should also mention that I have a dual boot system with Windows/Linux Mint.

And finally, the RDR2 version is the one from the Epic Games Store.
Importing the game into a new profile works normally; the launcher, Social Club, and an extra package are installed, which is the source of the controversy: VulkanRT-1.1.108.0-Installer.exe, also included in the /RedDeadRedemption2/Redistributables/ directory. This package installs several Vulkan libraries in the /windows/system32 directory of the prefix, in addition to registering the runtime.

So, to import/install on an existing prefix, you have to force the installation of this package. I did this by adding a new game, running the installer first, and then running VulkanRT-1.1.108.0-Installer.exe. (Trying it from within the game's installation/import didn't work; it wouldn't install.) After installation, I closed the process without finalizing it.

Result: The game runs with Vulkan without any errors.

Before this, with the default prefix, I encountered errors such as: Multiple instances when installing the Rockstar launcher, the well-known Social Club error 18, or the game needing Steam (even though it's an Epic Games title). Basically, if it wasn't one thing, it was another.

In short: If this error occurs, forcing the installation of the VulkanRT-1.1.108.0-Installer.exe package with the corresponding prefix was the solution for me.

I hope this helps, and sorry for the long post 0:)

Cheers!

Edit: My native language is Spanish, and this is a Google translation, so I hope the spirit of the post is preserved :P

**Comment by srusterv** [comment]
I literally cannot believe that this is the one and only place I've personally been able to find this solution - and it's dead simple and in retrospect seems so obvious, and it is the ONLY thing that worked for me.

On Heroic Launcher, I literally just opened the RDR2 settings, clicked "Run EXE on prefix" and then opened that included vulkan installer and let it run. Instantly solved the problem. Even less hassle than messing with the settings file, and brings the benefits of Vulkan for performance, mods, reshade, etc. Thank you @makimanson for posting this because I was trying to figure out if I could mess with vulkan layers and all sorts of other shenanigans and this was dead simple and worked.

**Reply by makimanson** [reply]
I'm glad to hear it was helpful. :)

**Reply by LinuxAndLucy** [reply]
OMFG, thank you.  I can now continue my 82% completed save...

If anybody else stumbles across this, do as @srusterv mentions above, but remember to remove the environment variable "Variable Name : WINEDLLOVERRIDES Value: vulkan-1=n,b" if you were using it prior.
